fix: tv login navigation

lancechant opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Chores**
  - Updated multiple dependencies to newer versions, including React, React Native, Expo, and related libraries.
  - Removed unused dependencies and added new ones for improved compatibility and features.
  - Improved Metro bundler configuration with better type safety and TV-specific file resolution.
  - Adjusted autolinking behavior for certain native modules based on the target platform (TV or non-TV).
  - Removed a plugin configuration for Google Cast from the app settings.

- **Refactor**
  - Streamlined import statements and improved type handling in several files.
  - Simplified color extraction logic for images by using direct imports.

- **Style**
  - Removed unused imports and made minor code organization changes for clarity.

No user-facing features or bug fixes were introduced in this release.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

lancechant committed:
fix: made tv os compile

Made useDownload return no-op functions instead of throwing an error
when running tvos

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
fix: home screen navigation

Removing the download icon from the header since not needed for TV's

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
fix: navigation issues with TV

upgraded to sdk 53 with expo
Used a google cast fix version that's compatible with sdk 53
Removed react-native-volume-manager as seems to break tv navigation

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Merge remote-tracking branch 'origin/develop' into fix/tv-login-navigation

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
recreated bun.lock file

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
added cross-env back

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
new bun.lock file

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
fixing tv build

fixed the application by removing react-native-google-cast from the
app.json as it's not needed for TV

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Removing usage of react-native-volume-manager

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
fixed react-native-volume-manager tv issues

Added a react-native config file that doesn't link if running for tv
added some dependencies for zeego that were needed
Uncommented code that used the volume manager

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Fixed android release build

Removed a package causing conflicts

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

Gauvino commented on package.json:
```diff
@@ -17,121 +17,99 @@
-    "typescript": "~5.7.3"
+    "react-test-renderer": "18.3.1",
+    "typescript": "~5.8.3"
   },
```
@lancechant check this also

lancechant replied:
@Gauvino changed the version back to `19.0.0`

lancechant committed:
Fixing react test renderer version

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Fixing react test renderer version + bun.lock

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Merge branch 'fix/tv-login-navigation' of https://github.com/lancechant/streamyfin into fix/tv-login-navigation

lancechant committed:
Merge remote-tracking branch 'origin/develop' into fix/tv-login-navigation

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Adding patch files

Temporalily adding patch files to get around build issue till a release
is done

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Updating uitextview to 2.0.4

The new version might fix the build issue with IOS

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
Merge remote-tracking branch 'origin/develop' into fix/tv-login-navigation

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

Gauvino commented on package.json:
```diff
@@ -18,96 +18,85 @@
+    "@react-navigation/material-top-tabs": "7.2.14",
     "@react-navigation/native": "^7.0.14",
-    "@shopify/flash-list": "1.8.3",
+    "@shopify/flash-list": "^2.0.0-rc.4",
```
Is the RC really needed can we just ping to 1.8.3 and wait for 2.0 to be released

lancechant replied:
Yes, it can't build otherwise

Gauvino commented on package.json:
```diff
@@ -18,96 +18,85 @@
-    "@futurejj/react-native-visibility-sensor": "^1.3.10",
+    "@bottom-tabs/react-navigation": "^0.9.0",
+    "@expo/metro-runtime": "~5.0.4",
+    "@expo/react-native-action-sheet": "^4.1.0",
```
Why downgrading this ?

lancechant replied:
Seems like it's from the constant merge conflicts with renovate PR's getting merged, will fix

Gauvino commented on package.json:
```diff
@@ -18,96 +18,85 @@
-    "expo-updates": "~0.26.17",
-    "expo-web-browser": "~14.0.2",
-    "i18next": "^25.0.0",
+    "cross-env": "^7.0.3",
```
keep cross-env in dev dep

Gauvino commented on package.json:
```diff
@@ -117,25 +106,22 @@
     "postinstall-postinstall": "^2.1.0",
-    "react-test-renderer": "19.0.0",
-    "typescript": "~5.7.3"
+    "react-test-renderer": "18.3.1",
```
Why downgrading this ?

Gauvino reviewed (changes requested):
Your removing many package, i've not tested it but is it really necessary ?

lancechant committed:
Fixing comments from code-rabbit

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant commented on biome.json:
```diff
@@ -1,14 +1,16 @@
 {
-  "$schema": "https://biomejs.dev/schemas/2.0.5/schema.json",
+  "$schema": "https://biomejs.dev/schemas/1.9.4/schema.json",
+  "organizeImports": {
```
@Gauvino I had to revert this with renovate as was breaking things, was excluding normal files

Gauvino replied:
I will investigate that

lancechant commented:
> Your removing many package, i've not tested it but is it really necessary ?

From my testing and build back up, some of the dependencies weren't needed from what I saw, from the testing I ran, things seem to be working, but not sure if some were removed and are needed, haven't seen that with the testing i've done

lancechant committed:
Merge remote-tracking branch 'origin/develop' into fix/tv-login-navigation

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>
